module-1/pandas-project/your-code/data-wrangling.ipynb

```diff
@@ -42,7 +42,7 @@
   },
```

Comment:
1. You say in the cell above cell to In[6] "I printed the summary information to further understand the composition of the data set". You only are considering the info function that gives you info on technical characteristics of the dataset (data types, number of records, number of nulls per column.....). It's good to run also the describe function(if you have various numeric variables in your dataset).
2. Really good practice to check in each chunk of code that the results are the ones you expected beforehand. 
Well done with that "I proceded to confirm that the data types were actually changed"
3. Really good hypothesis. Further analyzing the data, I discovered that 'Pop. Density (per sq. mi.)' comes from diving 'Area (sq. mi.)' by 'Population'. Since this is a data point we could later get if necesary for analysis, I decided to drop the column. In real world, you would need to come back from further steps to check whether your initial hypothesis still holds.  
4. In [13]: You can change the order and the name of columns in a more readable way when you are dealing with a high number of columns/variables. You can run a loop for do this stuff in an automatic way and also loc and iloc pandas functions. Take a look on these when you have a free minute. 
5. You say. Replacing Values "Since I want to do an analysis between 'Literacy (%)' and 'GDP ($ per capita)' and 'Literacy (%)' had 18 nulls, I decided to replace the nulls for the mean.I found the mean for 'Literacy (%)' first." Well done not to drop nulls or fill with 0s. Really good approach the one you have taken. I would like to explain a couple of things on this when you have time we can seat. Just for you to explore, you can do the bining first and after calculate the average per bin, take a look on the type and number of outliers. If you have many outliers or very extreme values the mean could not be the best measure to fill nulls. Am I explaining properly?
6. I encourage you to finish the lab because you have done a really good exercise.